docs/provision-elms-with-terraform.md

```diff
@@ -0,0 +1,91 @@
+# Provision the ELMS infrastructure using Terraform
+
+[Terraform](https://www.terraform.io/) can be used for managing the ELMS infrastructure on Azure. You can find the definitions in the [terraform](../terraform) folder.
+Terraform can be set up locally and resources can be deployed to a chosen Azure subscription.
+The following steps are needed to achieve this:
+
+## 1. Prepare environment
+
+### Option 1: Use Docker devcontainer
+
+1. Install [Docker Desktop](https://docs.docker.com/desktop/)
+2. Install Visual Studio extension [Remote - Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
+3. Open project in a devcontainer. Run the `Remote-Containers: Open Folder in Container...` command and select the local folder
+
+After that you will have all the needed tools, so continue with [step 3](#3-create-the-terraform-state-storage).
+
+### Option 2: Install Terraform and Azure CLI manually
+
+1. [Install Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+2. [Install Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
+
+## 2. Log into Azure CLI
+
+Open a terminal, run the following commands and, if needed, select the Azure subscription which will be targeted by the Terraform deployment:
+
+```shell
+az login
+az account set --subscription="<subscription_id>"
+```
+
+## 3. Create the Terraform state storage
```

Comment:
I think it's important to note here that using a backend state storage is optional.  If the user doesn't want to manage the Azure resources with Terraform beyond their creation, they can also just run the commands locally without any remote storage

Reply:
We will document that it is optional.